src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -14,9 +15,9 @@ namespace NuGet.Protocol
     /// </summary>
     public class MetadataReferenceCache
     {
-        private readonly Dictionary<string, string> _stringCache = new Dictionary<string, string>(StringComparer.Ordinal);
-        private readonly Dictionary<Type, PropertyInfo[]> _propertyCache = new Dictionary<Type, PropertyInfo[]>();
-        private readonly Dictionary<string, NuGetVersion> _versionCache = new Dictionary<string, NuGetVersion>(StringComparer.Ordinal);
+        private readonly ConcurrentDictionary<string, string> _stringCache = new ConcurrentDictionary<string, string>(StringComparer.Ordinal);
```

Comment:
I think the common scenario is having 1 page and thus 1 thread, so making this concurrent dictionary is an overkill. 
concurrent dictionary has 5-10x the overhead of a regular dictionary with a lock.